### PR TITLE
fix(context): makes `on` default to shadow-private

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. made `@colorContextConsumer` default to shadow-private
1. updated changesets

## Testing Instructions

1. check stat, alert, spinner in DP

### CODE REVIEW

**Review by zeroedin — CHANGES_REQUESTED** [review]
Added a PR to this PR that addresses some issues I came across.

**Comment by bennypowers** [comment]
@zeroedin I encorporated the demo you wrote in #678 here. I didn't add the other `color-palette` themes because that's a bigger design question (and because fixes made here to blockquote seem to solve the problem)

**Review by marionnegp — APPROVED** [review]
Color context changes are working, and the component styling looks correct to me.

File: elements/rh-context-provider/rh-context-provider.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
-:host { display: block; }
-div {
+:host {
```
Comment (zeroedin) [inline elements/rh-context-provider/rh-context-provider.css:1]: I still find myself questioning if this component should provide the background color.

Are we creating a component that would be better named`rh-surface` or `rh-band` by doing this?  Should this component be a physical (for lack of better terms) component that modifies the visual styles or is it only providing context to child components?

```html
/** Mobile Screen size  (Simplified example) **/
<rh-secondary-nav color-palette="dark">
  /* Shadow DOM */
  <nav class="compact">
    <div id="container" class="expanded">
      ...
      <div id="cta">
        <rh-context-provider color-palette="lightest">
          <slot name="cta"></slot>
        </rh-context-provider>
      </div>
    </div>
  </nav>
</rh-secondary-nav>
```

```css
#container.expanded #cta {
  background-color: var(--rh-color-surface-lightest, #ffffff);
}
```

The above example shows how `rh-secondary-nav` implements the background color when a mobile menu is expanded.   The CSS provides the surface color while the `rh-context-provider` only provides the context to the child consumer components.  The mobile menu toggle event modifies the context provider with the switch of contexts.    

I guess my question is about which component should own those styles?  Should this component remain transparent and only provide context?  If not should we rename it?

**Review by zeroedin — APPROVED** [review]
Llamas Graze The Mountain

File: elements/rh-context-provider/rh-context-provider.css (hunk 1 above)
Comment (bennypowers) [reply elements/rh-context-provider/rh-context-provider.css:1]: I'm ok with removing the background colors. @zeroedin can you make that change? Make sure to check all the demo files

**Pull request merged**